docs/madr/decisions/050-mesh-passthrough.md

```diff
@@ -0,0 +1,130 @@
+# MeshPassthrough
+
+* Status: accepted
+
+## Context and Problem Statement
+
+During the design process of `MeshExternalService`, we discovered that allowing traffic to specific domains and wildcard domains (e.g., `*.aws.us-east-2.com`) is a great functionality. However, this capability doesn't fully fit within the `MeshExternalService` concept. `MeshExternalService` is a kind that allows adding some external services to work within your mesh, be targeted by policies, and be configured by them. Unfortunately, when a user needs to communicate with services that shouldn't be modified by Envoy, we discovered that we need a separate policy to handle them.
+
+## Current solution
+
+Currently, we don't support passthrough mode. Kuma creates DNS entries for real domains and returns a custom IP. This has a limitation: you cannot have two external services using the same domain. Additionally, we don't support wildcard domains.
+
+## Considered Options
+
+* MeshExternalService resource with passthrough mode support
+* New MeshPassthrough policy
+
+## Decision Outcome
+
+New `MeshPassthrough` policy seems like cleaner, more flexible.
+
+### Positive Consequences
+
+* More flexibility: We can target specific proxies.
+* Clarity: There is no confusion about which fields are involved in the policy configuration.
+* Simplicity: Policies are cleaner and cannot be configured with other policies, unlike MeshExternalService.
+* No confusion with Universal: There is no issue with non-transparent proxies.
+
+### Negative Consequences
+
+* Additional complexity: Introduction of another policy.
+* Resource management: The need to manage two separate resources.
+
+### New MeshPassthrough policy
+
+We would like to introduce a new policy `MeshPassthrough` which allows exposing domains, IPs, CIDRs through specific sidecars. Policy should be only configured by the MeshOperator and allowed in `kuma-system`.
+
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshPassthrough
+metadata:
+  name: custom-passthrough
+  namespace: kuma-system
+  labels:
+    kuma.io/mesh: default
+spec:
+  targetRef:
+    kind: MeshSubset # Mesh, MeshSubset
+    tags:
+      chatgpt.io/access: "true"
+  default:
+    matchAppend:
+    - type: Domain
+      value: api.chatgpt.com
+      port: 443
+      protocol: tls    
+```
+
+`MeshPassthrough` should allows targeting specific subset of proxies and apply configuration only on them. We should support following kinds: `Mesh`, `MeshSubset` and in the future `MeshService`.
```

Comment:
```suggestion
`MeshPassthrough` should allow targeting specific subset of proxies and apply configuration only on them. We should support following kinds: `Mesh`, `MeshSubset` and in the future `MeshService`.
```

Reply:
I wouldn't allow `MeshService` people should you dataplane tags